crates/oci/src/loader.rs

```diff
@@ -0,0 +1,127 @@
+use std::path::{Path, PathBuf};
+
+use anyhow::{anyhow, ensure, Context, Result};
+use oci_distribution::Reference;
+use reqwest::Url;
+use spin_app::locked::{ContentPath, ContentRef, LockedApp, LockedComponent};
+use spin_loader::cache::Cache;
+
+use crate::{Client, ORIGIN_URL_SCHEME};
+
+/// OciLoader loads an OCI app in preparation for running with Spin.
+pub struct OciLoader {
+    working_dir: PathBuf,
+}
+
+impl OciLoader {
+    /// Creates a new OciLoader which builds temporary mount directory(s) in
+    /// the given working_dir.
+    pub fn new(working_dir: impl Into<PathBuf>) -> Self {
+        let working_dir = working_dir.into();
+        Self { working_dir }
+    }
+
+    /// Loads a LockedApp with the given OCI client and reference.
+    pub async fn load_app(&self, client: &mut Client, reference: &str) -> Result<LockedApp> {
+        // Fetch app
+        client
+            .pull(reference)
+            .await
+            .context("cannot pull Spin application from registry")?;
+
+        // Read locked app
```

Comment:
I don't think you use the path or content again after this section - suggest extracting a `read_locked_app` function